### PR TITLE
Share length validation

### PR DESCRIPTION
**Pull request by MatheusFranco99**
# Overview

This PR adds a validation to the runners' construction function (except by `CommitteeRunner`) to assert that the `share` argument has only one entry (only one share).

### CODE REVIEW

File: ssv/aggregator.go [hunk 1]
```diff
@@ -29,7 +29,12 @@ func NewAggregatorRunner(
 	operatorSigner *types.OperatorSigner,
 	valCheck qbft.ProposedValueCheckF,
 	highestDecidedSlot phase0.Slot,
-) Runner {
+) (Runner, error) {
+
+	if len(share) != 1 {
+		return nil, errors.New("must have only one share")
```
Comment (GalRogozinski) [inline ssv/aggregator.go:35]: if we add an error we should have an accompaning test

Comment (MatheusFranco99) [reply ssv/aggregator.go:35]: On it

**Review by GalRogozinski — APPROVED** [review]

**Review by y0sher — APPROVED** [review]
Approving this but with an *.

* @GalRogozinski @MatheusFranco99 
I think if this check is required maybe its time to think of separating the base for committee and other runners. Since other runners will never recv more than one share, it looks weird that their func signature can recv a map of shares.

**Comment by MatheusFranco99** [comment]
@y0sher I agree.
I have opened an issue for that to be done later on

File: ssv/spectest/tests/runner/construction/many_shares.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+	"github.com/ssvlabs/ssv-spec/types"
+	"github.com/ssvlabs/ssv-spec/types/testingutils"
+)
+
+func ManyShares() tests.SpecTest {
+
+	ks := testingutils.KeySetMapForValidators(10)
+	shares := testingutils.ShareMapFromKeySetMap(ks)
+
+	// No errors since one share must be valid for all runners
```
Comment (alan-ssvlabs) [inline ssv/spectest/tests/runner/construction/many_shares.go:14]: Is this comment right for this test?

File: ssv/spectest/tests/runner/construction/no_shares.go [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+	"github.com/attestantio/go-eth2-client/spec/phase0"
+	"github.com/ssvlabs/ssv-spec/ssv/spectest/tests"
+	"github.com/ssvlabs/ssv-spec/types"
+)
+
+func NoShares() tests.SpecTest {
+
+	shares := map[phase0.ValidatorIndex]*types.Share{}
+
+	// No errors since one share must be valid for all runners
```
Comment (alan-ssvlabs) [inline ssv/spectest/tests/runner/construction/no_shares.go:13]: and here

**Review by alan-ssvlabs — APPROVED** [review]

File: ssv/spectest/tests/runner/construction/many_shares.go (hunk 2 above)
Comment (MatheusFranco99) [reply ssv/spectest/tests/runner/construction/many_shares.go:14]: Indeed not. Updating. Thanks for the catch!

File: ssv/spectest/tests/runner/construction/no_shares.go (hunk 3 above)
Comment (MatheusFranco99) [reply ssv/spectest/tests/runner/construction/no_shares.go:13]: Updating, thanks!

**Pull request merged**